request_test.go

```diff
@@ -53,3 +54,11 @@ func testDecodable(t *testing.T, name string, out decoder, in []byte) {
 		t.Error("Decoding", name, "failed:", err)
 	}
 }
+
+func testDecoder(t *testing.T, name string, e encoder, d decoder, bytes []byte) {
```

Comment:
this tests both encoding and decoding, so a name like `testEncodeDecode` seems more appropriate